### ISSUE TITLE
project specific preferences for the node.js area are not really usable.

### ISSUE THREAD

**Issue by jcompagner** [angelozerr/typescript.java#115]
enable project specific settings on the node.js tab is not really usable because it does store values that are really os specific  for example i see:

nodeJSEmbedded=node-v4.2.4-win32-x86_64
nodeJSPath=C\:\\Program Files\\nodejs\\node.exe

both of them really would not work for mac or linux if i pushed that.

I think the nodeJSEmbedded should really be just the node version: node-v4.2.4
an then it should map on the actual os specific installed
or is that version really important? Can this not just be a boolean true/false to use the embedded?

The nodeJSPath is only useable if that would be a eclipse variable that a user has to configure first.

But i guess the question is more what would this node.js project specific setting really solve?
When do we want to really push on the project level something like this?

**Comment by angelozerr** [comment]
> both of them really would not work for mac or linux if i pushed that.

The combo of nodejs must be filled according the OS, no?

> I think the nodeJSEmbedded should really be just the node version: node-v4.2.4
an then it should map on the actual os specific installed

Yes you are right.

> or is that version really important? 

No 

> Can this not just be a boolean true/false to use the embedded?

Other plugins could extend the embedded nodejs combo, so the boolean is a bad idea.

> But i guess the question is more what would this node.js project specific setting really solve?

Today nothing, but perhaps in the future TypeScript version will require some node version.

> When do we want to really push on the project level something like this?

If one day TypeScript requieres a version of node, in this case you will have the capability to override the nodejs.

**Comment by angelozerr** [comment]
@jcompagner could you answered me please. Thanks!

**Comment by jcompagner** [comment]
> The combo of nodejs must be filled according the OS, no?

project preferences, those are stored in a settings file in the project so is shared across all team members. Those can be in a mixed environment (so linux,mac and windows) So those preferences should be in my eyes os independent, else it is of no use to share them or give them that option.

**Comment by angelozerr** [comment]
> So those preferences should be in my eyes os independent

It works like this, no?

**Comment by jcompagner** [comment]
yes for the embedded it uses now:

useNodeJSEmbedded=true

so that is better i guess, then it uses what ever it is installed

The only thing is that when using an external one (a path) that is then really installation depending. So using that in the project settings is just not really handy or useful.

**Comment by angelozerr** [comment]
> The only thing is that when using an external one (a path) that is then really installation depending. 

Ok, I understand you wish to have the capability to declare your path with ${project_loc:myproject)/node

is that?

**Comment by jcompagner** [comment]
yes or a plain eclipse variable, so they should configure this once.
But not that important. i think if you want to use it as a path then you don't make this a project preference setting..

**Comment by angelozerr** [comment]
@jcompagner for your information, now I store:

`nodeJSEmbedded=node-v6.9.4`
